src/main/java/de/hpi/bpt/argos/api/entity/EntityEndpointImpl.java

```diff
@@ -0,0 +1,332 @@
+import java.util.List;
+import java.util.Set;
+
+import static spark.Spark.halt;
+
+/**
+ * {@inheritDoc}
+ * This is the implementation.
+ */
+public class EntityEndpointImpl implements EntityEndpoint {
+
+    private static final Logger logger = LoggerFactory.getLogger(EventTypeEndpointImpl.class);
+    private static final Gson serializer = new Gson();
+    private static final RestEndpointUtil endpointUtil = RestEndpointUtilImpl.getInstance();
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setup(Service sparkService) {
+        sparkService.get(EntityEndpoint.getEntityBaseUri(), this::getEntity);
+        sparkService.get(EntityEndpoint.getChildEntitiesBaseUri(), this::getChildEntities);
+        sparkService.get(EntityEndpoint.getEventTypesOfEntityBaseUri(), this::getEventTypesOfEntity);
+        sparkService.get(EntityEndpoint.getEventsOfEntityBaseUri(), this::getEventsOfEntity);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getEntity(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        long entityId = getEntityId(request);
+        Entity eventType = PersistenceAdapterImpl.getInstance().getEntity(entityId);
+        List<Attribute> attributes = PersistenceAdapterImpl.getInstance().getAttributes(entityId);
+        JsonObject jsonEventType = getEntityJson(eventType, attributes);
+
+        response.body(serializer.toJson(jsonEventType));
+        endpointUtil.logSendingResponse(logger, request, response.status(), response.body());
+        return response.body();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getChildEntities(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        long entityId = getEntityId(request);
+        long entityTypeId = getEntityTypeId(request);
+        List<Entity> childEntities = PersistenceAdapterImpl.getInstance().getEntities(entityId, entityTypeId);
+        List<Attribute> attributesToInclude = getAttributesToInclude(request);
+        JsonObject jsonEntities = getEntitiesJson(childEntities, attributesToInclude);
+
+        response.body(serializer.toJson(jsonEntities));
+        endpointUtil.logSendingResponse(logger, request, response.status(), response.body());
+        return response.body();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getEventTypesOfEntity(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        long entityId = getEntityId(request);
+        List<EventType> allEventTypes = PersistenceAdapterImpl.getInstance().getEventTypes();
+
+        Set<EventType> eventTypes = new HashSet<>();
+        for (EventType eventType : allEventTypes) {
+            String sql = getExistsEventQuery(entityId, eventType.getId());
+            if (PersistenceAdapterImpl.getInstance().getExistsEvent(sql)){

```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') '{' is not preceded with whitespace. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=checkstyle%3AWhitespace_Around_BP2016)